### PR TITLE
Bubble up hint exceptions in hooks, so the user can see the hint

### PR DESCRIPTION
**Pull request by butonic**
Otherwise the exception gets swallowed in the hook catch all and the user just does not even see an error.

@MorrisJobke @DeepDiver1975 thougts?

### CODE REVIEW

**Comment by PVince81** [comment]
@nickvergessen as discussed ?

File: lib/private/hook.php [hunk 1]
```diff
@@ -112,6 +112,9 @@ static public function emit($signalClass, $signalName, $params = []) {
 				if (empty($message)) {
 					$message = get_class($e);
 				}
+				if($e instanceof \OC\HintException) {
+					throw $e;
+				}
```
Comment (nickvergessen) [inline lib/private/hook.php]: Should still log?

Comment (MorrisJobke) [inline lib/private/hook.php]: @butonic I would also like to log the exception. Simply move these lines after the logging right below would solve this ;)

Comment (butonic) [inline lib/private/hook.php]: Well, if it is a hint exception, ERROR might be too hard

Comment (MorrisJobke) [inline lib/private/hook.php]: Then log it as warning. I would still love to see this in logs :)

**Comment by DeepDiver1975** [comment]
@icewind1991 @PVince81 @schiesbn @nickvergessen @blizzz please review - THX

**Comment by icewind1991** [comment]
:+1: looks good

**Comment by MorrisJobke** [comment]
:+1:

**Pull request merged**